File: backend/internal/model/blessing.go
package model

import "time"

// Blessing 祝福留言实体，挂在心愿留言板上，可携带虚拟礼物表情。
type Blessing struct {
	ID            uint64    `gorm:"primaryKey" json:"id"`
	WishID        uint64    `gorm:"index;not null" json:"wish_id"`
	UserID        uint64    `gorm:"index;not null" json:"user_id"`
	Content       string    `gorm:"type:text;not null" json:"content"`
	GiftEmoji     string    `gorm:"size:50" json:"gift_emoji"`
	IsCelebrating bool      `gorm:"not null;default:false" json:"is_celebrating"`
	CreatedAt     time.Time `json:"created_at"`
}

File: frontend/src/api/blessing.ts
import { http } from "@/utils/request";
import type { PageResult } from "./wish";

export interface Blessing {
  id: number;
  wish_id: number;
  user_id: number;
  sender_name?: string;
  content: string;
  gift_emoji?: string;
  is_celebrating: boolean;
  created_at: string;
}

export const blessingApi = {
  create: (wishId: number, payload: { content: string; gift_emoji?: string }) =>
    http.post<Blessing>(`/wishes/${wishId}/blessings`, payload),
  list: (wishId: number, params?: Record<string, string | number | undefined>) =>
    http.get<PageResult<Blessing>>(`/wishes/${wishId}/blessings`, params),
};
